server/core/src/initServer.cpp

```diff
@@ -133,15 +95,12 @@ namespace
             r.mtime(SET_TIME_TO_NOW_KW);
         }
 
-        if (cond_input.contains(CHKSUM_KW)) {
-            r.checksum(cond_input.at(CHKSUM_KW).value());
-        }
-
+        r.checksum(""); // TODO Should the checksum be cleared?
```

Comment:
@trel Should the checksum field be erased if the client does not call close? Could do this based on whether bytes were written.

Reply:
If bytes were written, we definitely clear...

if no bytes were written... I think it would be okay to leave it as it was... could that be misleading in any case?

Reply:
counter question... any benefit to leaving it as is...?

Reply:
Erasing the checksum if bytes were written seems safe. Any operation that truncates a replica on open will automatically erase the checksum too, which is good.

As for the counter question, I don't know if there are benefits to leaving the checksum as is. I'm leaning towards it being more confusing.

Reply:
i think so too.   in an error case, i think we should torch it - nobody will have a false positive that way.  and worst case, they have to recalculate on a known good replica.

Reply:
What do you mean by error case?

The latest changes and tests show that the agent can erase the checksum based on whether bytes were written or not. Is that still what we want or did we want to always erase the checksum if the client doesn't call close?

Reply:
sorry, not error case... just unknown reason for 'lack of close'.

i think we agreed if the client doesn't call close on a write operation, they don't get a checksum generated, so... we should clear the checksum, regardless.  it can't be misleading that way.

Reply:
Ok. I'll update the code to always erase the checksum.